presto-main/src/test/java/com/facebook/presto/operator/BenchmarkMoreByteArrays.java

```diff
@@ -0,0 +1,154 @@
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkMoreByteArrays
+{
+    @Benchmark
+    public void fillToByteArray(BenchmarkData data)
+    {
+        fill(data.bytes, 0, data.bytes.length, (byte) 5);
+    }
+
+    @Benchmark
+    public void fillToBasicSliceOutput(BenchmarkData data)
+    {
+        data.basicSliceOutput.reset();
+        for (int i = 0; i < data.bytes.length; i++) {
+            data.basicSliceOutput.writeByte((byte) 5);
+        }
+    }
+
+    @Benchmark
+    public void fillToDynamicSliceOutput(BenchmarkData data)
+    {
+        data.dynamicSliceOutput.reset();
+        for (int i = 0; i < data.bytes.length; i++) {
+            data.dynamicSliceOutput.writeByte((byte) 5);
+        }
+    }
+
+    @Benchmark
+    public void setBytesToByteArray(BenchmarkData data)
+    {
+        setBytes(data.bytes, 0, data.byteValues, 0, data.bytes.length);
+    }
+
+    @Benchmark
+    public void setBytesToBasicSliceOutput(BenchmarkData data)
+    {
+        data.basicSliceOutput.reset();
+        data.basicSliceOutput.writeBytes(data.slice, 0, data.slice.length());
+    }
+
+    @Benchmark
+    public void setBytesToDynamicSliceOutput(BenchmarkData data)
+    {
+        data.dynamicSliceOutput.reset();
+        data.dynamicSliceOutput.writeBytes(data.slice, 0, data.slice.length());
+    }
+
+    @Benchmark
+    public void setIntsToByteArray(BenchmarkData data)
+    {
+        setInts(data.bytes, 0, data.intValues, 0, data.POSITIONS_PER_PAGE);
+    }
+
+    @Benchmark
+    public void setIntsToBasicSliceOutput(BenchmarkData data)
+    {
+        data.basicSliceOutput.reset();
+        for (int i = 0; i < data.POSITIONS_PER_PAGE; i++) {
+            data.basicSliceOutput.writeInt(data.intValues[i]);
+        }
+    }
+
+    @Benchmark
+    public void setIntsToDynamicSliceOutput(BenchmarkData data)
+    {
+        data.dynamicSliceOutput.reset();
+        for (int i = 0; i < data.POSITIONS_PER_PAGE; i++) {
+            data.dynamicSliceOutput.writeInt(data.intValues[i]);
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private static final int POSITIONS_PER_PAGE = 10000;
+
+        private final Random random = new Random(0);
+
+        private final int[] intValues = IntStream.range(0, POSITIONS_PER_PAGE).toArray();
+        private final byte[] byteValues = new byte[POSITIONS_PER_PAGE * ARRAY_LONG_INDEX_SCALE];
+        private final Slice slice = Slices.wrappedBuffer(byteValues);
+
+        private final byte[] bytes = new byte[POSITIONS_PER_PAGE * ARRAY_LONG_INDEX_SCALE];
+        private final SliceOutput basicSliceOutput = Slices.wrappedBuffer(new byte[POSITIONS_PER_PAGE * ARRAY_LONG_INDEX_SCALE]).getOutput();
+        private final SliceOutput dynamicSliceOutput = new DynamicSliceOutput(POSITIONS_PER_PAGE * ARRAY_LONG_INDEX_SCALE);
+
+        @Setup
+        public void setup()
+        {
+            random.nextBytes(byteValues);
```

Comment:
nit: any reason why `byteValues` are generated randomly while `intValues` are from in range `(0, POSITIONS_PER_PAGE)` ?

Reply:
I was just making them non-zero since the values don't matter. To make them agree with each other, I changed the intValues to be random too:

`private final int[] intValues = IntStream.range(0, POSITIONS_PER_PAGE).map(i -> random.nextInt()).toArray();`